### ISSUE TITLE
Add tutorial

### ISSUE THREAD

**Issue by arximboldi** [arximboldi/mixco#6]
As Be-ing put it very well in a comment:

> I like this idea, but the barrier to entry to working with Mixco is quite high. The way the documentation is now requires a strong background in Node, JS, CoffeeScript, and Mixxx's mapping formats to know where to start. The documentation for each of Mixco's modules is pretty good. But what I think is missing is a general overview tutorial that explains how to start working with Mixco from the beginning and how the different parts of Mixco work together.

I agree that such tutorial should be added.

**Comment by Be-ing** [comment]
To make this accessible to people without strong JS/CoffeeScript backgrounds, I suggest explaining some aspects of the language or linking to documentation that does. For JavaScript, the [Mozilla Developer Network](https://developer.mozilla.org/en-US/docs/Web/JavaScript) is a good reference.

**Comment by ppwfx** [comment]
Is there a way to use it without a controller?

**Comment by arximboldi** [comment]
What do you mean "use it without a controller"?

If you are thinking of using this for writing scripts for Mixxx without having a controller: yes, you can. To interact with it you need to use MIDI though. You can send MIDI back and forth to an external script to interact with the mouse/keyboard or some other thing. You could even write a browser script using WebMIDI to create an UI for your Mixxx script.

**Comment by ppwfx** [comment]
yeah that's what I meant.. I new to music / midi stuff.. using the keyboard to create midi signal sounds like fun.. can you provide me some links to get started? will contribute an example implementation if that's of interest for this project

**Comment by arximboldi** [comment]
Cool...  So from your Github I infer that you are already good at JavaScript and Node.js.

In JavaScript, there are two ways you can do MIDI from outside of Mixxx

* **Node.js**
    https://github.com/justinlatimer/node-midi

* **Browser**
    https://www.w3.org/TR/webmidi/
    https://github.com/cotejp/webmidi
    https://www.npmjs.com/package/web-midi-api

Back in the days I wrote some experiments using the browser MIDI API in Chrome. You can find a few tutorials by Googling for "WebMIDI tutorials". Note that you will need to create a virtual MIDI port so Mixxx and your app can talk through it. This is easy on Linux and Mac, not sure about Windows.

Cheers!

**Comment by ppwfx** [comment]
Great man! Thanks a lot, I'm looking forward for the weekend :)
